### DISCUSSION TITLE
Trigger rule for mixed results?

### DISCUSSION THREAD

**Discussion by scottbw** [apache/airflow#32914] [Q&A]
Hi fellow airflowers(?),

I have a set of mapped tasks, and what I want to happen next depends on the outcome of those tasks.

I have three possible downstream tasks, and I'm trying to define which ones are scheduled using trigger rules.

1. [all completed, none failed] -> Normal flow, everything is fine, lets go.
2. [all failed] -> Error flow, notify operator
3. [some succeeded, some failed] -> Uh oh, we're now in an unstable state. I need to try to rollback 

So the first two are fine but I'm struggling to figure out the third condition. None of the trigger rules seem to cover it. Any suggestions?

Thanks!

**Comment by potiuk** [comment]
you can use `all_done` and use the output they send via xcom as status to see which failed. It's difficult to generalise such rule - 2 failed? 3 failed? 1 skipped 1 failed? Especially that you might want to react quicker - one_failed for example will trigger immediately when one of the tasks failed - regardless if other completeda - and this is much more needed for triggering rule, because it might move the workflow faster. 

When you want to wait for all tasks to complete and then check their status "all_done" is likely the best. Just make sure your task outputs are set in xcome as needed and then you can use branch or other similar operators to decide what to do if some of them failed.

**Reply by scottbw** [reply]
Thanks Jarek, I'll try that out!
